File: src/main/java/io/github/malczuuu/iemu/domain/StateService.java
package io.github.malczuuu.iemu.domain;

import java.time.Instant;
import java.util.function.Consumer;

public interface StateService {

  StateDTO getState();

  void changeState(StateDTO state);

  void subscribeOnCurrentTimeChange(Consumer<Instant> consumer);

  void subscribeOnStateChange(Consumer<Boolean> consumer);

  void subscribeOnDimmerChange(Consumer<Integer> consumer);

  void subscribeOnTimeCounterChange(Consumer<Long> consumer);

  void resetErrors();

  void shutdown();
}

File: src/main/kotlin/io/github/malczuuu/iemu/domain/StateService.kt
package io.github.malczuuu.iemu.domain

import java.time.Instant

interface StateService {

  fun getState(): State

  fun changeState(update: StateUpdate)

  fun subscribeOnCurrentTimeChange(consumer: (Instant) -> Unit): Boolean

  fun subscribeOnStateChange(consumer: (Boolean) -> Unit): Boolean

  fun subscribeOnDimmerChange(consumer: (Int) -> Unit): Boolean

  fun subscribeOnTimeCounterChange(consumer: (Long) -> Unit): Boolean

  fun resetErrors()

  fun shutdown()
}
